.github/workflows/release.yml

```diff
@@ -2,380 +2,413 @@ name: WasmCloud Release
+          --override-input secret-key-base 'file:///dev/stdin' \
+          --override-input hostcore_wasmcloud_native-x86_64-pc-windows-msvc /tmp/nif \
+          -L .#${{ matrix.package }}-burrito-x86_64-windows
+
+    - uses: actions/upload-artifact@v3
+      with:
+        name: ${{ matrix.package }}_x86_64_windows
+        path: ./result/bin/${{ matrix.package }}_x86_64_windows.exe
+        if-no-files-found: error
+
+  test-darwin:
+    runs-on: macos-latest
+    needs:
+    - burrito-darwin
+    steps:
+    - run: curl -sL 'https://github.com/nats-io/nats-server/releases/download/v2.9.16/nats-server-v2.9.16-darwin-amd64.tar.gz' | tar xzv --strip-components 1 
+    - uses: actions/download-artifact@v3
+      with:
+        name: host_core_x86_64_darwin
+    - run: chmod +x ./host_core_x86_64_darwin
+    - run: |
+        ./nats-server &
+        ./host_core_x86_64_darwin & sleep 30; kill $!
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: wasmcloud_host_x86_64_darwin
+    - run: chmod +x ./wasmcloud_host_x86_64_darwin
+    - run: |
+        ./nats-server &
+        ./wasmcloud_host_x86_64_darwin & sleep 30; kill $!
+
+  test-linux:
+    runs-on: ubuntu-latest
+    needs:
+    - burrito-linux
+    services:
+      nats:
+        image: nats
+        ports:
+        - 4222:4222
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: host_core_x86_64_linux_gnu
+    - run: chmod +x ./host_core_x86_64_linux_gnu
+    - run: ./host_core_x86_64_linux_gnu & sleep 30; kill $!
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: wasmcloud_host_x86_64_linux_gnu
+    - run: chmod +x ./wasmcloud_host_x86_64_linux_gnu
+    - run: ./wasmcloud_host_x86_64_linux_gnu & sleep 30; kill $!
+
+  test-windows:
+    runs-on: windows-latest
+    needs:
+    - burrito-windows
+    steps:
+    - run: curl -sLo 'nats.tar.gz' 'https://github.com/nats-io/nats-server/releases/download/v2.9.16/nats-server-v2.9.16-windows-amd64.tar.gz'
+    - run: tar xzvf nats.tar.gz --strip-components 1
+    - uses: actions/download-artifact@v3
+      with:
+        name: host_core_x86_64_windows
+    - run: |
+        Start-Process ./nats-server.exe
+        Start-Process ./host_core_x86_64_windows.exe
+        timeout /t 30
+        Taskkill /im host_core_x86_64_windows.exe /f
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: wasmcloud_host_x86_64_windows
+    - run: |
+        Start-Process ./nats-server.exe
+        Start-Process ./wasmcloud_host_x86_64_windows.exe 
+        timeout /t 30
+        Taskkill /im wasmcloud_host_x86_64_windows.exe /f
+
+  release:
+    # TODO: Build on PRs and pushes to main (but do not push)
+    # This is currently blocked on the `app-version` computation, which only supports semver tags
+    # E.g.:
+    # ```
+    # Error: buildx failed with: ERROR: invalid tag "wasmcloud.azurecr.io/wasmcloud_host:616/merge": invalid reference format
+    # ```
```

Comment:
Leaving comment for later, if we want to run canary releases or something this sounds good. We can create an issue